core/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java

```diff
@@ -176,14 +175,8 @@ public void onRemoval(RemovalNotification<Object, Cache<Query, Value>> notificat
         }
 
         for (Map.Entry<Query, Value> entry : value.asMap().entrySet()) {
```

Comment:
unrelated to your change, but maybe iterate over `.values()` given that we never read `entry.getKey()`?